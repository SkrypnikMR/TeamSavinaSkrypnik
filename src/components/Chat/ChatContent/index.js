import { connect } from 'react-redux';
import { userEmail } from '/src/Store/user/selectors';
import { currentRoomName } from '/src/Store/chat/selectors';
import ChatContent from './ChatContent.jsx';

const mapStateToProps = state => ({
    currentRoomName: currentRoomName(state),
    currentUser: userEmail(state),
});

export default connect(mapStateToProps)(ChatContent);
