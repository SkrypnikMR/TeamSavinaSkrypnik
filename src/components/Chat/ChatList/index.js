import { connect } from 'react-redux';
import ChatList from './ChatList.jsx';
import { rooms } from '/src/Store/chat/selectors';
import { changeModalVisibility } from '/src/Store/modals/actions';

const mapStateToProps = state => ({ rooms: rooms(state) });
const mapDispatchToProps = dispatch => ({
    changeModalVisibility: payload => dispatch(changeModalVisibility(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChatList);
