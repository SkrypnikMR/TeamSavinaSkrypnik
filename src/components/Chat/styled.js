import styled from 'styled-components';
import { textColorBlack } from '/src/components/UI/baseLayout';

export const StChat = styled.div`
    display: flex;
    width: auto;
    margin: 0;
    justify-content: space-between;
    align-items: space-between;
    color: ${({ color = textColorBlack }) => color};
    font-family: 'Play', sans-serif;
    border-radius: 0px;
    position: relative;
    overflow:  hidden;
    touch-action: none;
    height: 90vh;
    overflow-x: hidden;
    overflow-y: no-scroll;
`;
