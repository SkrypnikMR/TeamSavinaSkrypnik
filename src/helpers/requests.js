import { routes } from '../constants/routes';

const { baseUrl } = routes;

export const getRequest = async (url, optionalHeaders = {}) => {
  const headers = Object.assign({}, optionalHeaders);
  const options = { method: 'GET', headers };
  const answer = await fetch(`${baseUrl}${url}`, options);
  if (!answer.ok) throw new Error(await answer.json());
  const result = await answer.json();
  return result;
};

export const postRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'POST',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};

export const putRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'PUT',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(`${baseUrl}${url}`, options);
  const result = await answer.json();
  return result;
};
export const createAuthHeader = (support) => { return { Authorization: support.getCookie('token') }; };
