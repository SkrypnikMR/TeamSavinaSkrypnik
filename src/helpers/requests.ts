import { routes } from '../constants/routes';
import { TUrl, TBody } from './types/requestTypes';

const { baseUrl } = routes;

export const getRequest = async (url: TUrl) => {
  const options = { method: 'GET' };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};

export const postRequest = async (url : TUrl, body : TBody) : Promise<Response> => {
      const options = {
              method: 'POST',
              body: JSON.stringify(body),
              headers: { 'Content-Type': 'application/json' },
      };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};
