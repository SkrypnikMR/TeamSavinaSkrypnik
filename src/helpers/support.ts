import moment from 'moment';

export const support = {
    setSessionStorageItem: (name, data) => {
        if (typeof data !== 'string') data = JSON.stringify(data);
        localStorage.setItem(name, data);
    },
    killSessionStorageItem: (name) => {
        localStorage.removeItem(name);
    },
    getSessionStorageItem: (name) => {
        const item = localStorage.getItem(name);
        return item?.charAt(0) === '{' ? JSON.parse(item) : item;
    },
    getMessagesFolders: (rooms) => {
        const messagesFolders = {};
        rooms.forEach((room) => {
            const { room_name } = room;
            messagesFolders[room_name] = [];
        });
        return messagesFolders;
    },
    getFormatedDate: date => moment(date).format('YYYY-MM-DD HH:mm:ss'),
    playAudio: (path) => {
        const audio = new Audio(path);
        audio.play();
    },
};
