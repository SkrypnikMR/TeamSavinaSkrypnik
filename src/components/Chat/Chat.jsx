import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import ChatList from './ChatList';
import ChatContent from './ChatContent';
import { StChat } from './styled';

const Chat = ({ userInit, userToken, init }) => {
  useEffect(() => {
    if (!userInit && userToken) init();
  }, [userInit]);
  return (
    <StChat>
      <ChatList />
      <ChatContent />
    </StChat>
  );
};

Chat.propTypes = {
  userInit: PropTypes.bool,
  userToken: PropTypes.any,
  init: PropTypes.func,
};

export default Chat;
