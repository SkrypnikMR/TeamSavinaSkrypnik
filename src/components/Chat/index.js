import { connect } from 'react-redux';
import Chat from './Chat.jsx';
import { userInit, userToken } from '../../Store/user/selectors.js';
import { init } from '../../Store/chat/actions';

const mapStateToProps = state => ({
    userInit: userInit(state),
    userToken: userToken(state),
});
const mapDispatchToProps = dispatch => ({ init: () => dispatch(init()) });

export default connect(mapStateToProps, mapDispatchToProps)(Chat);
