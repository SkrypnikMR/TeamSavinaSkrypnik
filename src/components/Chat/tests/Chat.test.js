import React from 'react';
import configureStore from 'redux-mock-store';
import { shallowSmart, mountSmart } from '/src/helpers/testHelper';
import Chat from '../Chat';

const mockStore = configureStore();
const store = mockStore({
  chat: {
    onlineUsers: [],
    messages: {},
    newMessage: '',
    rooms: [],
    currentRoom: { room_id: null, room_name: '' },
    error: false,
    isLoading: false,
  },
  user: {
    userInfo: {
      email: 'SkripnikMRW@gmail.com',
      firstName: 'Max',
      lastName: 'Skripnik',
    },
  },
});

describe('Chat', () => {
  const init = jest.fn();
  it('Should match snapshot', () => {
    const component = shallowSmart(<Chat init={init} />, store);
    expect(component.html).toMatchSnapshot();
  });
  it('Should render StChat', () => {
    const component = mountSmart(<Chat init={init} />, store);
    expect(component.find('styled__StChat')).toHaveLength(1);
  });
  it('Should render ChatList', () => {
    const component = mountSmart(<Chat init={init} />, store);
    expect(component.find('ChatList')).toHaveLength(1);
  });
  it('Should render ChatContent', () => {
    const component = mountSmart(<Chat init={init} />, store);
    expect(component.find('ChatContent')).toHaveLength(1);
  });
});
