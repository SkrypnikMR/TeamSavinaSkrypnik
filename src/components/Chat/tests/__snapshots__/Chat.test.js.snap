// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Should match snapshot 1`] = `[Function]`;
